src/index.ts

```diff
@@ -176,10 +186,79 @@ class Input {
     }
 }
 
+class Client {
+    tox: Tox
+    commander: Commander
+    chatWith: number | null
+
+    constructor() {
+        const tox = new Tox()
+
+        tox.addEventListener("core", ((ev: CustomEvent<ToxEvent>) => {
+            this.onToxEvent(ev.detail)
+        }) as EventListener)
+
+        this.tox = new Tox()
+        this.commander = new Commander()
+        this.chatWith = null
+    }
+
+    runCommand(command: string) {
+        const action = this.commander.evaluate(command)
+
+        if(action === null) { return }
+
+        let response = null
+        switch (action.action) {
+            case "info":
+                response = this.tox.info()
+                break;
+            case "add":
+                if(action.message !== null) {
```

Comment:
No need to check it because you check in inside .addFriend

Reply:
null ≠ undefined

Reply:
Then what's the point of `message?: string`?

Reply:
You can use `tox.addFriend(toxId)` instead of `tox.addFriend(toxId, null)`.

Reply:
> null ≠ undefined

What's the point then?

Reply:
In the world of js there's no point. I could try using `undefined` instead of `null` in `Action` though.